File: youlexuan-shop-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {
    this.delete = function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    };

    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?tbid="+id);
    };

    this.add = function (entity) {
        return $http.post("../brand/add.do",entity);
    };

    this.update = function (entity) {
        return $http.post("../brand/update.do",entity);
    };

    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    }

    this.findPage = function (page,rows) {
        return $http.get("../brand/findPage.do?pageNum="+page+"&pageSize="+rows) ;
    }

    this.search = function (pageNum,pageSize,searchEntity) {
        return $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);
    }
});

File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/BrandService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbBrand;

import java.util.List;

public interface BrandService {
    /**
     * 查询所有的品牌列表，不带分页
     * @return
     */
    public List<TbBrand> findAll();


    /**
     * 分页查询
     * @param pageNum  第几页
     * @param pageSize  每页显示多少条
     * @return
     */
    public PageResult findPage(int pageNum , int pageSize);


    /**
     * 带有条件的分页查询
     * @param pageNum
     * @param pageSize
     * @param tbBrand
     * @return
     */
    public PageResult findPage(int pageNum , int pageSize,TbBrand tbBrand);




    /**
     * 新增品牌
     * @param brand
     */
    public void add(TbBrand brand);


    /**
     * 根据id查询
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);


    /**
     * 修改（根据id修改）
     * @param tbBrand
     */
    public void update(TbBrand tbBrand);


    /**
     * 删除
     * @param ids
     */
    public void delete(Long[] ids);

}
